sources/core/Stride.Core.Mathematics/Vector2.cs

```diff
@@ -207,6 +207,24 @@ public float[] ToArray()
             return new float[] { X, Y };
         }
 
+        /// <summary>
+        /// Moves the first vector2 to the second one in a straight line.
+        /// </summary>
+        /// <param name="First">The first point.</param>
+        /// <param name="Second">The second point.</param>
+        /// <param name="MaxDistancePerCall">The rate at which the first point is going to move towards the second point.</param>
+        public static Vector2 Goto(Vector2 First, Vector2 Second, float MaxDistancePerCall)
+        {
+            Vector2 Distance = Subtract(Second, First);
```

Comment:
here you could use the subtract with `ref/out` parameters to solve some copies.

Reply:
Shouldn't we use the `in` keyword and have only a single version of this function?

Reply:
From my understanding, it should be fine if we add `readonly` to `Length()`, which is the only non-field (property) accessed by the method.

Reply:
> Not sure about if compiler would make defensive copies for such a struct or not though, anybody knows?

It shouldn't trigger a defensive copy as long as we don't use struct properties, or at least that's how it should work, maybe they changed it in newer versions but I couldn't repro that quirk.
Benchmark with net6 showed no perf' difference between passing by ``in`` or by ``ref``, by value was significantly slower of course.
```cs
[Benchmark]
public void PassedByValue()
{
    Vector3 a = new Vector3( 10, 5, 9 ), b = new Vector3( 8, 6, -10 );
    for( int i = 0; i < 100000; i++ )
    {
        a = Vector3.MoveTo( a, b, 0.15f );
        a.X += 10f;
    }
}

[Benchmark]
public void PassedByRef()
{
    Vector3 a = new Vector3( 10, 5, 9 ), b = new Vector3( 8, 6, -10 );
    for( int i = 0; i < 100000; i++ )
    {
        Vector3.MoveTo( ref a, ref b, 0.15f, out a );
        a.X += 10f;
    }
}

[Benchmark]
public void PassedByIn()
{
    Vector3 a = new Vector3( 10, 5, 9 ), b = new Vector3( 8, 6, -10 );
    for( int i = 0; i < 100000; i++ )
    {
        a = Vector3.MoveToIn( a, b, 0.15f );
        a.X += 10f;
    }
}
```

**That said though, we have been going back and forth with @VexedUXR for a while now, it would be kind of rude to ask of him to change stuff again, one of us three should incorporate the rest of what's required and merge this in. I'll get to that tomorrow if neither of you had the time to do it by then** :)

If you want to take a look at the IL: 
https://sharplab.io/#v2:C4LglgNgPgAgTABgLACh4EZWpugnACgCJAeDcBB9wgSgG4sVsBmAAgGdgAnAVwGNhGA1AKY8A9m3qoA3qkYzZMmExwA2RmwEBDACbCAdhACejMDt4BlMAC8BASR0AhfcAHNGAXkb0AVM0sDhAM3x/CGF1YGpaOTkFRmVVDW09Q0ERMUYALQE2YTdGHQEAd34hYFF6fAi6FCjZGLi1LV0DYtSmAFUdMGAADVz8opTSsXx0ADoEfwAaRgRxqZm5yuka2MV0FQbE5sGyxg6ugE0+wpah8tmJ6bHLhYml6pW69fjGpNPd/eB044GSsvwLvNAVdFjQqitVrFnpsmsk/mkAPL5H7vYbXebokF3MHLGoAegJjAAPMwOABbMnqNj6AB8uKiBLxjAAKgALASMXpcYRkgAOugEJkYAUYwHZjAAbvDRvS5IziXjSRSqbTZbUmMFQrxujiHvjCSTyZTqXS9QzCWyOUduXyBUKRWKOVLUjKzXKDYqjSrTRCYpqwowDrqIfLDcqTWqZPLLRlGDb+fl7f5ReLnUNXSGPUrjaq3erGP7eOlg48mDthoXJeoIBwBBRI4wpHmar13BLq7WwRC5Ec2x2BF3uzJvn2awOGwBfSKl1HlSsAD2mlf0S5CAYs9ebjYbclbjHng6HvcY+kP3ZHjAsZ7kU7NDZiACNhMIIIxrMwAHKiSkQXyaBtNkOMgAOYCLwEixAA7IwACyYSsqMACCD7MPg+C9J4nIUIwADUjD4EcmEHNheH4N8mHpNhAC0jDoP42FEjM/hUIwt4QmxM6VmKYDMAA2sYvDGJoAjzgAugBO6yKBwCSTIgFAXIzAFF0XCsvhQkiZuClyPJ2lRFw6jMByCAgFBnLXnpcaGRy6CmTA0FBrJ3YGUZjBwHZ0HFk5N7TnpYrZEU/SMIhbDAeSgrAAiHCRf4ABK6g6KBACi85cAIvLAGAuhEBp86ENMhC2JoYBpS4/iiLOqgcDoBbZGSMyijkcDTMYXA1j4UqjJQFk1LeTlGTJW46d5shKSpan4LlWmWbplkyC5xmmXu7ZjixD4NAA1j12kLbRpnHitnaMOtGhbSNNS7e5sajkdJ3qGdQ1AcJ/jqBwECgCmAV5CcIVhWSEVRTF8WJQIKVpRlWU6DlOjCXlBVFSVzgFhV5ZMJwNX+HVDWlG5LU6G1HAdQIXWVHNHFDuTUR9VufprrwAAygrAWKFQSY98jQUEdMUHBYqjKYACObDAGhGFYbh+GEYGJH4eRGQUKTKzU76Gp04wjOJWKgscFSAiaKzW6zb6nNi90MsEYwRHm3LlHbROjD0veawqDAAAssHCFKzLCPgajJqjtU8tMfuVaUq5aowlLzsybDqFKEAACI8cACVpdMwjRZVaiku9001EbKwB8VrCp+O7OMKYHAPuw6g8L7AjJmH8TJpjQfCpnxcp/jdYlkOlYQEzYq5J3pejBrzOshUvfdmAyb4FHMdxwIifJ6XjA0u4A+a2pUBQIwW8T247gIHn2nZ29vDuKU21yMvRnnfEOeX99vytEEdWjL0eEj93n+MEyB8h6YQXrHeOScS7dyXB/I439V6/yOAAwealgHqGjqA5e4Cu5p0DmSUY3xYEQLSng/++8kGW0jqgxeYC4FpUVr1XyUQngqADjBT2Ahvb4ADq3Mk0wA5N0rCApeK9CF1jZgpAO593o31kKwr2PsQ7cODg3Rq0xBHUJEenTOkjwjSI5o/C+dsGHRGdpVWR7DhC2EmjVLhdU8ah2EOHAMaiME0NEYbJyRdXG6IrlXGuPBLH8Nse3XgP9aHTwhP3Mh7hQnE3Hize4ClZ74WccIrBHIN6kO3owXemTD6uGPqfBS9lGreLvgIB+xSgqo3fjyP+BC0l/0QVklBaChGYNLlA2pMDGAxNGAg3JQCKGtPUWkzpuD8E9NccQpph8WlUJcSIuhVMjH5jiG7Hx1dY51xDgHAe/hgBKP9vCNGYBgKsgOcErOzgL6FNkAXGo2iUTVL2cAP+NE2CnPOZ/aYLy+mMHeZ815BwfkN1ed8AFZywVLJ8uCGcaz3aV02bXYAljjCVReXYiRgLNG8Akdc3OYigKPPcFU45+Bfm9AhV87oIL9l/KpUC2lYL/mqEBXg6FshbwTiAA==

Reply:
Interesting, I though calling `Length()` would make it do defensive copies.
Totally agree with one of us doing it, sorry for chiming in late.

Reply:
Ah didn't know about this improvement, i thought it's nice to have it in sync with the other method. So we can also keep it as it's now and clean up the ref methods with all the others at a later point.

@VexedUXR so basically do with this info as you please :-)